spec/classes/redis_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'spec_helper'
```

Comment:
You could simplify your testing approach here by adding an assertion to https://github.com/theforeman/puppet-pulpcore/blob/master/spec/classes/pulpcore_spec.rb#L486 that `redis` is included, and then in the defaults (https://github.com/theforeman/puppet-pulpcore/blob/master/spec/classes/pulpcore_spec.rb#L8) that redis is not included. 

Then you could add a small block for the enabling of rq tasking system similar to https://github.com/theforeman/puppet-pulpcore/blob/master/spec/classes/pulpcore_spec.rb#L486 and include a similar assertion of redis.

In this way you'd keep all of the assertions logically together, and they would reflect the parameters being passed in and how that affects things. This method, while it does work, I think it's downside is it separates the expectations.

Reply:
After trying this out, I see what you mean. It's a very nice approach.

Updated with the change. There is no longer a separate `spec/classes/redis_spec.rb` -- now the cases of default arguments, RQ tasking system, and content caching, are all covered in the `spec/classes/pulpcore_spec.rb`